programs/drift_vaults/src/instructions/request_withdraw.rs

```diff
@@ -0,0 +1,119 @@
+use crate::constraints::{
+    is_authority_for_vault_depositor, is_user_for_vault, is_user_stats_for_vault,
+};
+use crate::error::ErrorCode;
+use crate::validate;
+use crate::{Vault, VaultDepositor};
+use anchor_lang::prelude::*;
+use anchor_spl::token::{self, Token, TokenAccount, Transfer};
+use drift::instructions::optional_accounts::{load_maps, AccountMaps};
+use drift::math::insurance::vault_amount_to_if_shares;
+use drift::math::margin::{
+    calculate_margin_requirement_and_total_collateral, MarginRequirementType,
+};
+use drift::program::Drift;
+use drift::state::perp_market_map::{get_writable_perp_market_set, MarketSet};
+use drift::state::user::User;
+
+pub fn request_withdraw<'info>(
+    ctx: Context<'_, '_, '_, 'info, RequestWithdraw<'info>>,
+    amount: u64,
```

Comment:
could be nice to let people withdraw using shares or token amount 🤔

Reply:
just add another param that is an enum, specifying whether request is in shares or token amount